src/helpers/helpers.ts

```diff
@@ -57,7 +58,28 @@ export async function verifyCamelKubernetesPluginIsInstalled(): Promise<boolean>
 	return output.includes('kubernetes');
 }
 
-async function runJBangCommandWithStatusBar(args: string, msg: string): Promise<string> {
+export async function getRuntimeInfo(): Promise<Runtime> {
+	const output = await runJBangCommandWithStatusBar('camel@apache/camel dependency runtime --json', 'Checking Project runtime...');
+
+	let runtime: Runtime = Runtime.NONE;
+
+	if (output.includes('camel-quarkus')) {
+		runtime = Runtime.QUARKUS;
+		// {"runtime":"camel-quarkus","camelVersion":"4.10.2","camelQuarkusVersion":"3.20.0","quarkusVersion":"3.23.0"}
+	} else if (output.includes('camel-spring-boot')) {
+		runtime = Runtime.SPRING_BOOT;
+		// {"runtime":"camel-spring-boot","camelVersion":"4.8.0.redhat-00022","springBootVersion":"3.3.6"}
+	}
+
+	return runtime;
+}
+
+async function runJBangCommandWithStatusBar(args: string, msg: string, options: ExecSyncOptions = {}): Promise<string> {
```

Comment:
for information, it is possible to provide the cwd as a parameter of the ExecSyncOptions (which would avoid the requirement to wait for https://github.com/apache/camel/pull/18308 )